tests/integration_tests/vectorstores/test_rocksetdb.py

```diff
@@ -0,0 +1,161 @@
+import os
+import logging
+import sys
+from typing import List
+
+import rockset
+import rockset.models
+import pytest
+
+from langchain.docstore.document import Document
+from tests.integration_tests.vectorstores.fake_embeddings import (
+    ConsistentFakeEmbeddings,
+    fake_texts,
+)
+from langchain.vectorstores.rocksetdb import RocksetDB
+
+logger = logging.getLogger(__name__)
+
+
+# To run these tests, make sure you have a collection with the name `langchain_demo`
+# and the following ingest transformation:
+#
+#   SELECT
+#       _input.* EXCEPT(_meta),
+#       VECTOR_ENFORCE(_input.description_embedding, 10, 'float') as description_embedding
+#   FROM
+#       _input
+#
+# We're using FakeEmbeddings utility to create text embeddings.
+# It generates vector embeddings of length 10.
+#
+# Set env ROCKSET_DELETE_DOCS_ON_START=1 if you want to delete all docs from
+# the collection before running any test.
+#
+# See https://rockset.com/blog/introducing-vector-search-on-rockset/ for more details.
+
+collection_name = "langchain_demo"
+text_key = "description"
+embedding_key = "description_embedding"
+
+
+class TestRockset:
+    rockset_vectorstore: RocksetDB
+
+    @classmethod
+    def setup_class(cls) -> None:
+        assert os.environ.get("ROCKSET_API_KEY") is not None
+        assert os.environ.get("ROCKSET_REGION") is not None
+
+        api_key = os.environ.get("ROCKSET_API_KEY")
+        region = os.environ.get("ROCKSET_REGION")
+        if region == "use1a1":
+            host = rockset.Regions.use1a1
+        elif region == "usw2a1":
+            host = rockset.Regions.usw2a1
+        elif region == "euc1a1":
+            host = rockset.Regions.euc1a1
+        elif region == "dev":
+            host = rockset.DevRegions.usw2a1
+        else:
+            logger.warn(
+                "Using ROCKSET_REGION:%s as it is.. \
+                You should know what you're doing...",
+                region,
+            )
+
+            host = region
+
+        client = rockset.RocksetClient(host, api_key)
+        if os.environ.get("ROCKSET_DELETE_DOCS_ON_START") == "1":
+            logger.info(
+                "Deleting all existing documents from the Rockset collection %s",
+                collection_name,
+            )
+
+            query_response = client.Queries.query(
+                sql={"query": "select _id from {}".format(collection_name)}
+            )
+            ids = [
+                str(r["_id"])
+                for r in getattr(
+                    query_response, query_response.attribute_map["results"]
+                )
```

Comment:
Are we just wanting to iterate through the list of docs here?

Reply:
If so, it would be simpler just to do: `for r in query_response.results`

Reply:
Ah, for some reason I though the mapping from `attribute_map` is essential, as that gives us the attribute name we want to read. But I suppose we can assume `results` and `_id` will be always named as such. Thanks @jhochmuth !